mitodl/mitxonline
# DRAFT: Update redux and redux-query packages to add more supported redux logic

# What are the relevant tickets?
supports 

# Description (What does it do?)
I've updated redux to 4.0.1 and redux-query to 3.3.1. The redux upgrade is just to support teh redux-query upgrade. The redux-query upgrade gives us redux-query hooks - less duplication of code + cleaner when trying to write wild async logic for the catalog.  We use the "advanced" middleware, so our implementation should just "work" since I removed the /advanced from the configureStore file

# How can this be tested?
Run all tests
Check that things function as-is.
Redux is used across the application to compose components mapping state and dispatch to props. If data loads, in theory it's working as intended.  Major uses include notifications, user data, catalog, enrollment data.

Look for console errors, missing data, and the like.

# Additional Context
[Info on redux-query 2->3](https://amplitude.github.io/redux-query/docs/upgrade-guides/v2-to-v3)
[Info on redux 3->4](https://medium.com/@vyakymenko/whats-new-in-redux-4-major-breaking-changes-4f22a25da921l)

Both upgrades don't look to touch anything that we're using.  Both packages had fairly minor upgrades at this point in general, so we're really checking to make sure no names changed on us.
